cups/file.c

```diff
@@ -335,6 +335,131 @@ _cupsFileCheckFilter(
 #endif /* !_WIN32 */
 
 
+/*
+ * 'cupsFileGetConfAndComments()' - Get line and comments from a configuration file.
+ */
+
+char *					/* O  - Line read or @code NULL@ on end of file or error */
+_cupsFileGetConfAndComments(cups_file_t *fp,	/* I  - CUPS file */
+                char        *buf,	/* O  - String buffer */
+		size_t      buflen,	/* I  - Size of string buffer */
+                char        **value,	/* O  - Pointer to value */
+		int         *linenum)	/* IO - Current line number */
+{
+  char	*ptr;				/* Pointer into line */
+
+
+ /*
+  * Range check input...
+  */
+
+  DEBUG_printf(("2cupsFileGetConfAndComments(fp=%p, buf=%p, buflen=" CUPS_LLFMT
+                ", value=%p, linenum=%p)", (void *)fp, (void *)buf, CUPS_LLCAST buflen, (void *)value, (void *)linenum));
+
+  if (!fp || (fp->mode != 'r' && fp->mode != 's') ||
+      !buf || buflen < 2 || !value)
+  {
+    if (value)
+      *value = NULL;
+
+    return (NULL);
+  }
+
+ /*
+  * Read the next line...
+  */
+
+  *value = NULL;
+
+  while (cupsFileGets(fp, buf, buflen))
+  {
+    (*linenum) ++;
+
+    /*
+     * Strip the comment from inline comment...
+     */
+     if ((ptr = strchr(buf, '#')) != NULL)
+     {
+       int index = (int) (ptr - buf);
+       while (index > 0 && _cups_isspace(buf[index]))
+         buf[index--] = '\0';
+     }
+
+     /*
+      * Strip leading whitespace...
+      */
+
+      for (ptr = buf; _cups_isspace(*ptr); ptr ++);
+
+      if (ptr > buf)
+        _cups_strcpy(buf, ptr);
+
+      /*
+       * See if there is anything left...
+       */
+
+       if (buf[0])
+       {
+         // return comment if any
```

Comment:
I know that I used it in my pseudocode, but on a second thought it looks out of place when all other one line comments are written as multiline comments. Would you mind turning it into multiline comment and rephrase the comment, so it looks like proper English instead of a personal note :D .